yjit/src/backend/ir.rs

```diff
@@ -1050,8 +1072,17 @@ impl Assembler
                             let reg = opnd.unwrap_reg();
                             Some(take_reg(&mut pool, &regs, &reg))
                         },
-                        _ => {
-                            Some(alloc_reg(&mut pool, &regs))
+                        _ => match alloc_reg(&mut pool, &regs) {
+                            Some(reg) => Some(reg),
```

Comment:
If you think it looks better, you could use `.or_else(|| ...)` or `unwrap_or_else` here.

Reply:
I actually started with `unwrap_or_else` code, but it causes "use of moved value `iterator.insns`". With `match`, you can move anything and then call `unreachable!` at the end, but with a closure passed to `unwrap_or_else`, calling `unreachable!` at the end of the closure doesn't allow you to move values used after `unwrap_or_else`.

We probably don't want to allow copying insns just for this purpose, and passing references out of an iterator instead of moving them seems also tricky with `iterator.insns`.